backend/coreapp/tests.py

```diff
@@ -33,7 +33,7 @@
 from .models.scratch import CompilerConfig, Scratch
 
 
-def requiresCompiler(*compilers: Compiler):
+def requiresCompiler(*compilers: Compiler) -> Callable[..., Any]:
```

Comment:
not that it matters whatsoever in a test, but you could do `Callable[[F], F]` where `F = TypeVar('F', bound=Callable[..., Any])` for more accurate typing. (The `bound=` part not too important.)